SSA/Projects/Tensor2D/Tensor2D.lean

```diff
@@ -0,0 +1,225 @@
+import SSA.Core.WellTypedFramework
+import Mathlib.Data.Matrix.Basic
+import Mathlib.Data.Nat.Basic
+import SSA.Core.Util
+
+
+/-- Tensor2d with existential dimension sizes. -/
+structure Tensor2d' (α : Type) where
+  n₀ : ℕ
```

Comment:
Can we use more descriptive names here? Like `dim1` and `dim2` or so? The `n` seems confusing enough, but then the names also don't match the count ;) it's called tensor 2d but I don't see any dimension called 2?